### PR TITLE
Feat/security

### PR DESCRIPTION
**Pull request by badmoodd**
Need discuss urls in loafLoader and add them to security config. This feat allows you to create user with initial genres. Also allows you to authorise in system and get token. 
Screenshot 2023-11-18 at 6 52 24 PM
Screenshot 2023-11-18 at 6 53 23 PM

### CODE REVIEW

File: src/main/java/io/github/semanticpie/derezhor/externalAgents/users/api/UserController.java [hunk 1]
```diff
@@ -19,7 +20,7 @@
 @AllArgsConstructor
 public class UserController {
 
-    private UserService userService;
+    private UserServiceImpl userService;
```
Comment (ardonplay) [inline src/main/java/io/github/semanticpie/derezhor/externalAgents/users/api/UserController.java:23]: Why?

**Review by ardonplay — APPROVED** [review]
It's cringe, but ok

File: src/main/java/io/github/semanticpie/derezhor/config/SecurityConfig.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        daoAuthenticationProvider.setUserDetailsService(userDetailsService);
+        return daoAuthenticationProvider;
+    }
+
+    @Bean
+    public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
+        return http.csrf(AbstractHttpConfigurer::disable)
+                .authorizeHttpRequests(
+                        (authorize) -> authorize
+                                .requestMatchers("/pie-tunes/api/v1/signup", "/pie-tunes/api/v1/auth", "/api/v1/derezhor/tracks").permitAll()
```
Comment (glebchanskiy) [inline src/main/java/io/github/semanticpie/derezhor/config/SecurityConfig.java:52]: Вынести бы в пропертяги

File: src/main/java/io/github/semanticpie/derezhor/externalAgents/users/api/UserController.java (hunk 1 above)
Comment (glebchanskiy) [reply src/main/java/io/github/semanticpie/derezhor/externalAgents/users/api/UserController.java:23]: Потому что!

File: src/main/resources/application.properties [hunk 3]
```diff
@@ -1,11 +1,13 @@
 server.port=8080
 derezhor.sc-machine.url=ws://127.0.0.1:8090
 derezhor.cache.config=cache.config.yml
+security.jwt.token.secret-key=8bd0df6428532f05f497f7a35f2434739ea89ef6b5c791f250
```
Comment (glebchanskiy) [inline src/main/resources/application.properties:4]: hack pentagon

**Review by glebchanskiy — APPROVED** [review]

**Pull request closed without merging**